.github/workflows/build-image.yml

```diff
@@ -0,0 +1,84 @@
+name: Build Container Image & Publish
+permissions: read-all
+on:
+  push:
+    # Publish semver tags as releases.
+    tags: [v*.*.*]
+
+env:
+  # Use docker.io for Docker Hub if empty
+  REGISTRY: ghcr.io
+  # github.repository as <account>/<repo>
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+      # This is used to complete the identity challenge
+      # with sigstore/fulcio when running outside of PRs.
+      id-token: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # Install the cosign tool except on PR
+      # https://github.com/sigstore/cosign-installer
+      - name: Install cosign
+        if: github.event_name != 'pull_request'
+        uses: sigstore/cosign-installer@v3
+
+      # Set up BuildKit Docker container builder to be able to build
+      # multi-platform images and export cache
+      # https://github.com/docker/setup-buildx-action
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      # Login against a Docker registry except on PR
+      # https://github.com/docker/login-action
+      - name: Log into registry ${{ env.REGISTRY }}
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Wouldn't this mean that the image will push via the user that merged the PR?

Reply:
The workflow will trigger only if a git tag is pushed in semver pattern or follows this regex pattern:
`/v\d+.\d+.\d+/gm`

Also, git tags created in a fork, do not get merged into main repo. 
https://stackoverflow.com/questions/12278660/adding-tags-to-a-pull-request/12279290